test/task.cc

```diff
@@ -168,11 +221,11 @@ TEST(Task, FailOnCallback) {
       ThrowsMessage<std::runtime_error>(StrEq("error from start")));
 }
 
-TEST(Task, FailTerminated) {
+TEST(Task, FailTerminatedPropagate) {
   MockFunction<void()> fail;
 
   EXPECT_CALL(fail, Call())
-      .Times(1);
+      .Times(0);
```

Comment:
Let's add `FAIL()` inside the eventual instead so that we are more explicit. I know we aren't doing that everywhere but I want to start doing that in our tests rather than counting using `.Times()`.